test/integration/shoots/creation/create_test.go

```diff
@@ -0,0 +1,224 @@
+	trueVar               = true
+	err                   error
+	gardenerTestOperation *GardenerTestOperation
+)
+
+const (
+	CreateAndReconcileTimeout = 7200 * time.Second
+	InitializationTimeout     = 20 * time.Second
+)
+
+func validateFlags() {
+	if !StringSet(*gardenerKubeconfigPath) {
+		Fail("you need to specify the correct path for the gardenerKubeconfigPath")
+	}
+
+	if !FileExists(*gardenerKubeconfigPath) {
+		Fail("gardenerKubeconfigPath path does not exist")
+	}
+
+	if !StringSet(*logLevel) {
+		level := "debug"
+		logLevel = &level
+	}
+
+	if !StringSet(*shootProviderType) {
+		Fail("you need to specify provider type of the shoot")
+	}
+
+	if !StringSet(*projectNamespace) {
+		Fail("you need to specify projectNamespace")
+	}
+
+	if StringSet(*shootMachineImageName) && !StringSet(*shootMachineImageVersion) {
+		Fail("shootMachineImageVersion has to be defined if shootMachineImageName is set")
+	}
+
+	if StringSet(*shootMachineImageVersion) && !StringSet(*shootMachineImageName) {
+		Fail("shootMachineImageName has to be defined if shootMachineImageVersion is set")
+	}
+
+	if !StringSet(*infrastructureProviderConfig) {
+		Fail(fmt.Sprintf("you need to specify the filepath to the infrastructureProviderConfig for the provider '%s'", *shootProviderType))
+	}
+
+	if !FileExists(*infrastructureProviderConfig) {
+		Fail("path to the infrastructureProviderConfig of the Shoot is invalid")
+	}
+
+	if StringSet(*controlPlaneProviderConfig) {
+		if !FileExists(*controlPlaneProviderConfig) {
+			Fail("path to the controlPlaneProviderConfig of the Shoot is invalid")
+		}
+	}
+
+	if StringSet(*networkingProviderConfig) {
+		if !FileExists(*networkingProviderConfig) {
+			Fail("path to the networkingProviderConfig of the Shoot is invalid")
+		}
+	}
+}
+
+var _ = Describe("Shoot Creation testing", func() {
+	CBeforeSuite(func(ctx context.Context) {
+		validateFlags()
+		testLogger = logger.AddWriter(logger.NewLogger(*logLevel), GinkgoWriter)
+
+		shoot := prepareShoot()
+		shootGardenerTest, err = NewShootGardenerTest(*gardenerKubeconfigPath, shoot, testLogger)
+		Expect(err).To(BeNil())
+		shootGardenerTest.SetupShootWorker(workerZone)
+		Expect(err).To(BeNil())
+		Expect(len(shootGardenerTest.Shoot.Spec.Provider.Workers)).Should(BeNumerically("==", 1))
+
+		// override default worker settings with flags
+		if shootMachineType != nil && len(*shootMachineType) > 0 {
+			shootGardenerTest.Shoot.Spec.Provider.Workers[0].Machine.Type = *shootMachineType
+		}
+
+		if shootMachineImageName != nil && len(*shootMachineImageName) > 0 {
+			shootGardenerTest.Shoot.Spec.Provider.Workers[0].Machine.Image.Name = *shootMachineImageName
+		}
+
+		if shootMachineImageVersion != nil && len(*shootMachineImageVersion) > 0 {
+			shootGardenerTest.Shoot.Spec.Provider.Workers[0].Machine.Image.Version = *shootMachineImageVersion
+		}
+
+		gardenerTestOperation, err = NewGardenTestOperation(ctx, shootGardenerTest.GardenClient, testLogger)
```

Comment:
context is not needed

Reply:
I vendored a new gardener-extension version to remove the call to this method